nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -843,6 +814,7 @@ in
 
         See the boot.initrd.secrets option documentation for more information.
       '')
+      (mkRemovedOptionModule [ "boot" "loader" "grub" "version" ] "Support for version 0.9x of GRUB was removed after being unsupported upstream for around a decade")
```

Comment:
The option could also have a type of `lib.types.enum [ 2 ]` while in the transition period.

Reply:
Scratch that, make it `types.enum [ 1 2 ]` and add a `config.assertion` that adds a user-actionable message.

Reply:
Oh, I didn't think of that… I'll implement one of the suggestions or try to come up with something else to not cause churn, once we figure out if we don't need to keep this to support the AWS overlords decisions.

Reply:
We probably will want to do `types.enum [ 1 2 ]` either way, but *somehow* only keep the config generation codepath alive for grub 1; e.g. still dropping all packages and all other use cases except config generation.

It might even be worth it to split the grub 1 and 2 config generations? I'm not even sure it really needed to be implemented through the same script, they are different enough that we should consider them two distinct bootloaders.